js/polyfill.js

```diff
@@ -1,12 +1,19 @@
 'use strict';
 
-(() => {
+// eslint-disable-next-line no-unused-expressions
+self.INJECTED !== 1 && (() => {
+
   if (!Object.entries) {
     Object.entries = obj => Object.keys(obj).map(k => [k, obj[k]]);
   }
   if (!Object.values) {
     Object.values = obj => Object.keys(obj).map(k => obj[k]);
   }
+
+  // the above was shared by content scripts and workers,
+  // the rest is only needed for our extension pages
+  if (!(self.chrome || 0).tabs) return;
```

Comment:
It is more clear to write `if (!self.chrome || !self.chrome.tabs) return;`

> like localStorage, which shouldn't be used by anything running in a worker

Yes, `localStorage` and `sessionStorage` are bound to the document and is not available in workers.

I do find that `stylelint` tried accessing `window.localStorage` but since it runs in Node.js, I think it can run without `localStorage` and `sessionStorage`.

Another way to load polyfills is to split DOM methods to another file e.g. `polyfill-dom.js` and only load it in our extension pages.

Reply:
Done.
I wish we were using a compiler so it'd be `!self.chrome?.tabs`